run_sharded_tests.sh

```
@@ -0,0 +1,256 @@
+set -euo pipefail
+
+# Check Bash version
+if [ "${BASH_VERSINFO[0]}" -lt 4 ] || { [ "${BASH_VERSINFO[0]}" -eq 4 ] && [ "${BASH_VERSINFO[1]}" -lt 3 ]; }; then
+    echo "Error: This script requires Bash 4.3+ for nameref support."
+    echo "Current version: ${BASH_VERSION}"
+    echo "On macOS, install via: brew install bash"
+    exit 1
+         exit 1
+     fi
+          exit 1
+     fi
```

Shellcheck (0.11.0)

[error] Parsing stopped here. Is this keyword correctly matched up?

(SC1089)